www/include/views/virtualMetrics/DB-Func.php

```diff
@@ -481,23 +485,33 @@ function &disableVirtualMetric($v_id = null, $force = 0)
     $repA = array("*", "+", "-", "?", "^", "$");
     $repB = array("\\\\*", "\\\\+", "\\\\-", "\\\\?", "\\\\^", "\\\\$");
     $l_where = ($force == 0) ? " AND `vmetric_activate` = '1'" : "";
-    $l_pqy = $pearDB->query("SELECT index_id, vmetric_name FROM `virtual_metrics` WHERE `vmetric_id`='$v_id'$l_where;");
-    if ($l_pqy->rowCount() == 1) {
-        $vmetric = $l_pqy->fetch();
-        $l_pqy->closeCursor();
-        $query = "SELECT vmetric_id FROM `virtual_metrics` WHERE `index_id`='" . $vmetric["index_id"] .
-            "' AND `vmetric_activate` = '1' " .
-            "AND `rpn_function` REGEXP '(^|,)" . str_replace($repA, $repB, $vmetric["vmetric_name"]) . "(,|$)';";
-        $l_pqy = $pearDB->query($query);
-        while ($d_vmetric = $l_pqy->fetch()) {
+    $statement = $pearDB->prepare(
+        "SELECT index_id, vmetric_name FROM `virtual_metrics` WHERE `vmetric_id`=:vmetric_id$l_where;"
+    );
+    $statement->bindValue(':vmetric_id', (int) $v_id, \PDO::PARAM_INT);
+    $statement->execute();
+    if ($statement->rowCount() == 1) {
+        $vmetric = $statement->fetch(\PDO::FETCH_ASSOC);
+        $statement->closeCursor();
+        $query = "SELECT vmetric_id FROM `virtual_metrics` WHERE `index_id`= :index_id AND `vmetric_activate` = '1' " .
+            "AND `rpn_function` REGEXP :rpn_function";
+        $statement = $pearDB->prepare($query);
+        $statement->bindValue(':index_id', (int) $vmetric["index_id"], \PDO::PARAM_INT);
+        $statement->bindValue(
+            ':rpn_function',
+            '(^|,)' . str_replace($repA, $repB, $vmetric["vmetric_name"]) . '(,|$)',
```

Comment:
I think you need to check that str_replace doesnt return an array before binding it.